### PR TITLE
Adds unit test for generic encoders (test_generic_encoders.py)

### PR DESCRIPTION
**Pull request by dantreiman**
Adds unit tests for:
- PassthroughEncoder
- DenseEncoder

from ludwig.encoders.generic_encoders

Also adds missing type annotations for parameters in test_category_encoders.py.

### CODE REVIEW

File: tests/ludwig/encoders/test_generic_encoders.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@pytest.mark.parametrize("input_size", [1, 2, 10])
+@pytest.mark.parametrize("num_layers", [1, 3, 6])
+@pytest.mark.parametrize("fc_size", [1, 2, 10, 256])
+def test_generic_dense_encoder(input_size: int, num_layers: int, fc_size: int):
+    dense_encoder = DenseEncoder(input_size, num_layers=num_layers, fc_size=fc_size)
+    inputs = torch.rand((10, input_size))
+    outputs = dense_encoder(inputs)
+    assert "encoder_output" in outputs
```
Comment (justinxzhao) [inline tests/ludwig/encoders/test_generic_encoders.py:30]: Does it make sense to simplify this to:

```
assert outputs["encoder_output"].shape[1:] == dense_encoder.output_shape
```

Failures due to a missing "encoder_output" on this simplified form are still quite descriptive.

**Review by justinxzhao — CHANGES_REQUESTED** [review]

File: tests/ludwig/encoders/test_generic_encoders.py (hunk 1 above)
Comment (dantreiman) [reply tests/ludwig/encoders/test_generic_encoders.py:30]: I prefer to err on the side of explicitness and 'one claim per assertion', but thats just my style preference I could go either way.

File: tests/ludwig/encoders/test_generic_encoders.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import pytest
+import torch
+
+from ludwig.encoders.generic_encoders import DenseEncoder, PassthroughEncoder
+
+
+@pytest.mark.parametrize("input_size", [1, 2, 10])
+@pytest.mark.parametrize("categorical", [True, False])
+def test_generic_passthrough_encoder(input_size: int, categorical: bool):
```
[the remark this answers is not in the document]
Comment (dantreiman) [reply tests/ludwig/encoders/test_generic_encoders.py:9]: Thats a convention I could get behind.  On the other hand, the Ludwig repo seems to use newlines the way a writer uses paragraphs, to group related ideas together - which is what I'm doing.

I think we should propose additions to the Ludwig style guide, and this discussion can happen at that level and then be applied to all PRs:  https://ludwig-ai.github.io/ludwig-docs/developer_guide/style_guidelines_and_tests/

File: tests/ludwig/encoders/test_generic_encoders.py (hunk 1 above)
Comment (justinxzhao) [reply tests/ludwig/encoders/test_generic_encoders.py:30]: The concision of having a single assert for one behavior seems nice, but I agree that it's also less explicit. Happy to go with what you and other people think is best here.

File: tests/ludwig/encoders/test_generic_encoders.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinxzhao) [reply tests/ludwig/encoders/test_generic_encoders.py:9]: Glad to hear that using newlines to structure tests seems like a reasonable convention!

In addition to better readability, I think it also encourages good unit test design by discouraging bad patterns where a certain phase (setup/exercise/verify) was overly complicated, like testing several different behaviors in the same test, for example.

I'll probably follow up with a PR to propose we adopt this convention. I don't think it's something we need to enforce religiously, but it's a useful framework for thinking about well-designed unit tests.

**Review by justinxzhao — APPROVED** [review]

**Pull request merged**